src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -182,25 +176,33 @@ export const selectAccountTypeByAccountId = createSelector(
     accountMetadata[accountId]?.accountType,
 )
 
-export const selectIsPortfolioLoaded = createSelector(
-  selectAccountSpecifiers,
-  selectPortfolioAssetIds,
-  (accountSpecifiers, portfolioAssetIds) => {
-    if (!accountSpecifiers.length) return false
-    /**
-     * for a given wallet - we can support 1 to n chains
-     * AppContext ensures we will have a portfolioAssetId for each chain's fee asset
-     * until the portfolioAssetIds includes supported chains fee assets, it's not fully loaded
-     * the golf below ensures that's the case
-     */
-    const assetIdToChainId = (assetId: AssetId): ChainId => fromAssetId(assetId).chainId
-
-    return !size(
-      difference(
-        uniq(map(accountSpecifiers, flow([keys, head]))),
-        uniq(map(portfolioAssetIds, assetIdToChainId)),
-      ),
-    )
+type PortfolioState = 'loading' | 'success' | 'error'
+
+type PortfolioStateGranular = {
+  [k: AccountId]: PortfolioState
+}
+
+export const selectPortfolioStateGranular = createDeepEqualOutputSelector(
+  selectPortfolioAccountMetadata,
+  selectPortfolioAccounts,
+  (accountMetadata, accountsById): PortfolioStateGranular => {
+    const requestedAccountIds = keys(accountMetadata)
+    return requestedAccountIds.reduce<PortfolioStateGranular>((acc, accountId) => {
+      const account = accountsById[accountId]
+      const accountState = account ? (account.assetIds.length ? 'success' : 'error') : 'loading'
+      acc[accountId] = accountState
```

Comment:
Suggestion: Would be more ergonomic to query `accountState.status` IMO, matching the status field in RTK / react-query and the like
```suggestion
      acc[accountId] = {status: accountState}
```